Vagrantfile

```diff
@@ -7,16 +7,52 @@ VAGRANTFILE_API_VERSION = "2"
 # Require a recent version of vagrant otherwise some have reported errors setting host names on boxes
 Vagrant.require_version ">= 1.6.2"
 
-if ARGV.first == "up" && ENV['USING_KUBE_SCRIPTS'] != 'true'
-  raise Vagrant::Errors::VagrantError.new, <<END
-Calling 'vagrant up' directly is not supported.  Instead, please run the following:
-
-  export KUBERNETES_PROVIDER=vagrant
-  export VAGRANT_DEFAULT_PROVIDER=providername
-  ./cluster/kube-up.sh
-END
+def providercheck
+    err = %{
+    'vagrant up' directly is not supported.  Instead, please run the following:
+    export KUBERNETES_PROVIDER=vagrant
+    export VAGRANT_DEFAULT_PROVIDER=providername
+    ./cluster/kube-up.sh
+    }
+    if ARGV.first == "up" && ENV['USING_KUBE_SCRIPTS'] != 'true'
+        return false, err
+    end
+    return true, nil
+end
+
+def binarycheck
+  if ARGV.first == "up"
+    version="_output/release-stage/full/kubernetes/version"
+    
+    # make sure there is a release...
+    err="No kube release found ! If running from a clone of the git repo, please run 'make quick-release'."
+    return false, err if ! File.exist?(version)
```

Comment:
Ohhhh I forgot about that use case.  Any clever generalized ideas on what to look for?